packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -1682,37 +1581,38 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
       throw SavedObjectsErrorHelpers.createGenericNotFoundEsUnavailableError();
     }
 
-    const typesAndSpaces = new Map<string, Set<string>>();
-    const spacesToAuthorize = new Set<string>([SavedObjectsUtils.namespaceIdToString(namespace)]); // Always check authZ for the active space
+    const authObjects = new Array<AuthorizeBulkGetObject>();
     const result = {
       saved_objects: expectedBulkGetResults.map((expectedResult) => {
         if (isLeft(expectedResult)) {
+          const { type, id } = expectedResult.value;
+          authObjects.push({ type, id, error: true });
           return expectedResult.value as any;
         }
 
         const {
           type,
           id,
-          namespaces = [SavedObjectsUtils.namespaceIdToString(namespace)], // set to default value for `rawDocExistsInNamespaces` check below
+          // set to default namespaces value for `rawDocExistsInNamespaces` check below
+          namespaces = [SavedObjectsUtils.namespaceIdToString(namespace)],
           esRequestIndex,
         } = expectedResult.value;
-        const doc = bulkGetResponse?.body.docs[esRequestIndex];
 
-        const spacesToEnforce =
-          typesAndSpaces.get(type) ?? new Set([SavedObjectsUtils.namespaceIdToString(namespace)]); // Always enforce authZ for the active space
-        for (const space of namespaces) {
-          spacesToEnforce.add(space);
-          typesAndSpaces.set(type, spacesToEnforce);
-          spacesToAuthorize.add(space);
-        }
+        const doc = bulkGetResponse?.body.docs[esRequestIndex];
 
         // @ts-expect-error MultiGetHit._source is optional
-        for (const space of doc?._source?.namespaces ?? []) {
-          spacesToAuthorize.add(space); // existing namespaces are included so we can later redact if necessary
-        }
+        const docNotFound = !doc?.found || !this.rawDocExistsInNamespaces(doc, namespaces);
 
-        // @ts-expect-error MultiGetHit._source is optional
-        if (!doc?.found || !this.rawDocExistsInNamespaces(doc, namespaces)) {
+        authObjects.push({
+          type,
+          id,
+          objectNamespaces: namespaces,
+          // @ts-expect-error MultiGetHit._source is optional
```

Comment:
just a note/comment: not something introduced by this PR and doesn't seem easy to fix so probably not worth fixing as part of this PR

... but the frequent use of `// @ts-expect-error MultiGetHit` concerns me because it ignores everything. In general we cannot be sure that we're handling `MgetMultiGetError` responses and in this specific case if `existingNamespaces` would ever not be optional typescript won't help us.

Reply:
Yes I agree, this is not ideal. I plan to make existingNamespaces not optional per [your comment below,](https://github.com/elastic/kibana/pull/148165#discussion_r1097304181) so I will have to update the assignment here.